core/middlewares/with-routes.ts

```diff
@@ -221,10 +226,12 @@ const getRouteInfo = async (request: NextRequest, event: NextFetchEvent) => {
   const channelId = request.headers.get('x-bc-channel-id') ?? '';
 
   try {
-    const pathname = clearLocaleFromPath(request.nextUrl.pathname, locale);
+    // For route resolution parity, we need to also include query params, otherwise certain redirects will not work.
+    const { pathname, search } = request.nextUrl;
+    const path = clearLocaleFromPath(pathname + search, locale);
```

Comment:
Just so we can limit the changes in this file, we can probably just inline this.
```suggestion
    // For route resolution parity, we need to also include query params, otherwise certain redirects will not work.
    const pathname = clearLocaleFromPath(request.nextUrl.pathname + request.nextUrl.search, locale);
```